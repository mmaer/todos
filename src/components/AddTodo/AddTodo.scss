.add-todo {
  position: relative;
  margin: 0;
  width: 100%;
  font-size: 24px;
  font-family: inherit;
  font-weight: inherit;
  border: none;
  padding: 16px 16px 16px 60px;
  box-shadow: inset 0 -2px 1px rgba(0, 0, 0, 0.03);
  box-sizing: border-box;
  background: rgba(0, 0, 0, 0.003);
}
