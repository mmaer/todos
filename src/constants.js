export const LOCAL_STORAGE_KEY = "todos";

export const ENTER_KEY = 13;
export const ESCAPE_KEY = 27;

export const FILTER_ALL = "all";
export const FILTER_ACTIVE = "active";
export const FILTER_COMPLETED = "completed";

export const FILTERS = [FILTER_ALL, FILTER_ACTIVE, FILTER_COMPLETED];
