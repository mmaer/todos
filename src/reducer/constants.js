export const ADD_TODO = "ADD_TODO";
export const REMOVE_TODO = "REMOVE_TODO";
export const EDIT_TODO_TITLE = "EDIT_TODO_TITLE";

export const TOGGLE_COMPLETED_STATUS = "TOGGLE_COMPLETED_STATUS";
export const TOGGLE_EDITING_STATUS = "TOGGLE_EDITING_STATUS";

export const REMOVE_COMPLETED_TODOS = "REMOVE_COMPLETED_TODOS";
