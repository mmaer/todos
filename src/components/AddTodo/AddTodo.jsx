import React, { useState, memo } from "react";

import { ENTER_KEY } from "../../constants";

import "./AddTodo.scss";

const AddTodo = memo(({ addTodo }) => {
  const [value, setValue] = useState("");

  const handleOnChange = ({ target }) => {
    setValue(target.value);
  };

  const handleEnter = event => {
    event.persist();
    if (event.which === ENTER_KEY) {
      if (value !== "") {
        addTodo({
          title: value,
          completed: false,
          editing: false
        });
        setValue("");
      }
    }
  };

  return (
    <input
      className="add-todo"
      placeholder="What needs to be done?"
      onKeyPress={handleEnter}
      onChange={handleOnChange}
      value={value}
    />
  );
});

export default AddTodo;
