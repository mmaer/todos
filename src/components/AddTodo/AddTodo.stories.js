import React from "react";

import AddTodo from ".";

export default {
  title: "AddTodo",
};

export const addTodo = () => <AddTodo />;
