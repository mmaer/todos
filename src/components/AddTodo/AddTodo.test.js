import React from "react";
import { shallow, mount } from "enzyme";

import { ENTER_KEY } from "../../constants";

import AddTodo from "./";

describe("AddTodo component", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<AddTodo addTodo={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should add new todo when enter key is pressed", () => {
    const mockAddTodoFunc = jest.fn();
    const wrapper = mount(<AddTodo addTodo={mockAddTodoFunc} />);
    const input = wrapper.find("input");

    input.simulate("change", { target: { value: "new-todo" } });
    input.simulate("keyPress", { persist: () => {}, which: ENTER_KEY });

    expect(mockAddTodoFunc).toBeCalledWith({
      completed: false,
      editing: false,
      title: "new-todo"
    });
  });

  it("Should not add new todo when value is equal ''", () => {
    const mockAddTodoFunc = jest.fn();
    const wrapper = mount(<AddTodo addTodo={mockAddTodoFunc} />);
    const input = wrapper.find("input");

    input.simulate("keyPress", { persist: () => {}, which: ENTER_KEY });

    expect(mockAddTodoFunc.mock.calls.length).toBe(0);
  });
});
