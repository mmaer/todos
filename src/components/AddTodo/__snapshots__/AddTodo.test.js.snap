// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo component Should match snapshot 1`] = `
<input
  className="add-todo"
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="What needs to be done?"
  value=""
/>
`;
